### ISSUE TITLE
Semantic Segmentation Training just hangs

### ISSUE THREAD

**Issue by BernhardGlueck** [PyTorchLightning/lightning-flash#928]
## 🐛 Bug

When training semantic segmentation ,the training process just hangs after a given amount of steps. 

I have tried various different heads/backbones, batch sizes, finetuning strategies, image sizes etc.. 

I also tried to set num_workers for the data loaders to 0 in order to isolate any problems with that.... Also no luck 

GPU(s) and CPU(s) are idle when this hang occurs. No activity recorded at all, no exceptions thrown from any process/subprocess

The hangs seems to occur after a similiar amount of iterations .. e.g with the full dataset the hang happens in epoch 10 ... 
If i limit the dataset to 1000 images  only then i can do 98 epochs before the training hang occurs.

This was tested on 
a) Bare Metal Ubuntu 20.04 with RTX 2080TI
b) WSL2 Ubuntu 20.04 with RTX 3090
c) Grid.ai with 1,2 and 4 V100 accelerators. 

### To Reproduce

Run the sample training script with the public dataset available here:
http://www.ipb.uni-bonn.de/data/sugarbeets2016/

#### Code sample
`import torch
import flash

from flash.image import SemanticSegmentation, SemanticSegmentationData
from pytorch_lightning import seed_everything

base_data_path = "/home/btest/work/datasets/sugarbeet_small/training/"
image_size = (640,480)
num_classes = 3
batch_size = 2
val_split = 0.2

source_path = base_data_path + "source"
mask_path = base_data_path + "mask"

seed_everything(42)

datamodule = SemanticSegmentationData.from_folders(
    train_folder=source_path,
    train_target_folder=mask_path,
    batch_size=batch_size,
    val_split=val_split,
    image_size=image_size,
    num_classes=num_classes
)

model = SemanticSegmentation(
    backbone="mobilenetv3_large_100",
    head="unetplusplus",
    num_classes=datamodule.num_classes,
)

trainer = flash.Trainer(max_epochs=100, gpus=torch.cuda.device_count())
trainer.finetune(model, datamodule=datamodule, strategy="freeze_unfreeze")`

### Expected behavior

Training runs to completion

### Environment

 - PyTorch Version (e.g., 1.0): 1.10
 - OS (e.g., Linux): Ubuntu 20.04
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.8
 - CUDA/cuDNN version: 11
 - GPU models and configuration: RTX 2080TI, V100, RTX 3090
 - Any other relevant information:

**Comment by BernhardGlueck** [comment]
I have narrowed the problem somewhat. The issue occurs only when cuda device is used, it works fine when using the CPU only.
